Use 'Evaluee/recipient' in student submission UI #8176

sukanta-27 opened the pull request:
Fixes #8176 

**Outline of Solution**

Moved the `Evaluee/Recipient` from `response.tag` to `questionWithResponses.tag` to avoid repeatative use of the word `Evaluee`, and modified the names of recipients to show in bold font.
<!-- Tell us how you solved the issue. -->
In addition to the UI changes shown in the screenshots in https://github.com/TEAMMATES/teammates/issues/8176#issuecomment-353736462, I also made the some changes in the UI of `instructorHelpSessions.jsp` to keep a consistent look all over the UI [Screenshot below].

**UI before:** 
![screenshot 275](https://user-images.githubusercontent.com/10681755/34328034-f3dda102-e8f9-11e7-8bfa-e7c17cccd2b0.png)

**UI after:**
![screenshot 280](https://user-images.githubusercontent.com/10681755/34410364-b15c7e30-ebf5-11e7-9b15-60f2b7d57641.png)


**In addition:**
I think we should change the name of the issue and PR to match the current scope of the PR, as we are not anymore replacing all `recipient` to `evaluee` everywhere in the UI, but rather are modifying the Student Submission forms primarily. (https://github.com/TEAMMATES/teammates/issues/8176#issuecomment-352323106)
<!-- If there are things you want the reviewers to focus on, include them here as well. -->
<!-- This portion can be skipped if the fix is trivial. -->

sukanta-27 commented:
Realised that moving the code for showing `Evaluee/Recipient`  from `response.tag` to `questionWithResponses.tag`, created the problem where `Evaluee/Recipient` is being used in places it is not supposed to. (because the `questionWithResponses.tag` is being used by every kind of questions). Screenshots below.

![screenshot 285](https://user-images.githubusercontent.com/10681755/34411575-795c0ec6-ebfd-11e7-9c63-24a3939455a3.png)
<br>
![screenshot 283](https://user-images.githubusercontent.com/10681755/34411582-87126d76-ebfd-11e7-90a1-2a310fb65510.png)
<br>
![screenshot 281](https://user-images.githubusercontent.com/10681755/34411591-90543cc0-ebfd-11e7-841b-353b7ac42725.png)

Looking for a way to solve this problem.
**In the meantime:** If I can get any **suggestions** or **hints** on how I can solve/approach this, will be very helpful as I am still a beginner at JSP and JSTL. (Though I know that finding how to solve is, is part of my expected work :sweat_smile: )

sukanta-27 commented:
The Current submission forms look like : 

**Instructor Feedback Submission Edit Page:**
![screenshot 289](https://user-images.githubusercontent.com/10681755/34829412-8e1142e2-f6d8-11e7-8460-3a7a57ed9852.png)

**Student Feedback Submission Edit Page:**

![screenshot 290](https://user-images.githubusercontent.com/10681755/34829618-2e487a14-f6d9-11e7-89cf-09448050c3c2.png)<br>

![screenshot 291](https://user-images.githubusercontent.com/10681755/34829620-350517ea-f6d9-11e7-808a-856f8d7fad4f.png)<br>
![screenshot 292](https://user-images.githubusercontent.com/10681755/34829630-3ace0560-f6d9-11e7-8094-ca63da527773.png)<br><br>

sukanta-27 commented:
@damithc Ready for review.

damithc commented:
@wkurniawan07 for your review please.

wkurniawan07 commented on src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag:
```diff
@@ -54,6 +56,20 @@
+      <c:if test="${not isRecipientNameHidden}">
+        <div class="col-sm-12 form-inline mobile-align-left">
+          <label for="input"
+              style="text-indent: 24px">
```
This looks short enough to be put in the previous line.

sukanta-27 replied:
Okay

wkurniawan07 commented on src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag:
```diff
@@ -54,6 +56,20 @@
+                  data-placement="top"
+                  title="<%= Const.Tooltips.EVALUEE_DESCRIPTION %>">
+                  Evaluee/Recipient
+              </span>
```
This `<span>` block should be pushed to the left by 2 spaces. Additionally, the `Evaluee/Recipient` text.

sukanta-27 replied:
Okay

wkurniawan07 commented on src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/response.tag:
```diff
@@ -26,29 +26,24 @@
+    <label>
+      <select class="participantSelect middlealign<c:if test="${not response.existingResponse}"> newResponse</c:if> form-control"
+          name="<%= Const.ParamsNames.FEEDBACK_RESPONSE_RECIPIENT %>-${questionWithResponses.question.qnIndx}-${response.responseIndx}"
+          ${isNumResponsesMax ? 'style="display:none;max-width:125px"' : 'style="width:275px;max-width:275px"'}
```
You can extract out `style=""` here and only put the conditional inside, i.e. `style="${...}"`.

sukanta-27 replied:
Okay.

wkurniawan07 commented on src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag:
```diff
@@ -7,6 +7,8 @@
 <%@ attribute name="isShowRealQuestionNumber" type="java.lang.Boolean" required="true" %>
 <%@ attribute name="isSessionOpenForSubmission" type="java.lang.Boolean" required="true" %>
 
+<c:set var="isRecipientNameHidden" value="${questionWithResponses.question.recipientNameHidden}"/>
```
Why is this the conditional used to determine whether the `Evaluee/Recipient` is displayed or not? What is the logic here?

sukanta-27 replied:
@wkurniawan07 
Without this conditional, `Evaluee/Recipient` was being displayed in Questions where there was no recipient name, like places like `Question 7` shown in the screenshot here : https://github.com/TEAMMATES/teammates/pull/8265#issuecomment-354286157, and other similar questions, and the UI did not look that much appealing for that reason.

wkurniawan07 replied:
That is not the issue. My question is why the condition to be used to determine whether `Evaluee/Recipient` is displayed is `questionWithResponses.question.recipientNameHidden`. How is that condition representative of whether or not to display `Evaluee/Recipient`?

sukanta-27 replied:
There was a similar test previously used in `response.tag` file, 
```jsp
${isRecipientNameHidden ?  'style="display:none"' : 'style="text-align:right"'}>
```
and I noticed that the questions which should have `Evaluee/Recipient` have recipient names like this one : 
![screenshot 295](https://user-images.githubusercontent.com/10681755/35054344-35682a4a-fba4-11e7-97f2-c61f0f0836e1.png)

And the questions where the `Evaluee/Recipient` does not look appealing or appropriate does not have recipient names like this one : 
![screenshot 281](https://user-images.githubusercontent.com/10681755/34411591-90543cc0-ebfd-11e7-841b-353b7ac42725.png)

So, I used `recipientNameHidden` variable to test if the recipient name is hidden in that question or not, and based on that I displayed the word `Evaluee/Recipient`.

wkurniawan07 commented on src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/questionWithResponses.tag:
```diff
@@ -54,6 +56,17 @@
+        <div class="col-sm-12 form-inline mobile-align-left">
+          <label for="input" style="text-indent: 24px">
+            <span data-toggle="tooltip" data-placement="top" title="<%= Const.Tooltips.EVALUEE_DESCRIPTION %>">
+                Evaluee/Recipient
```
Move this to the left by 2 spaces.

sukanta-27 replied:
Done.

damithc commented:
@sukanta-27 are the help pages updated to match this UI change?

sukanta-27 commented:
@damithc Sir,  I only encountered the word `Evaluee` in the Instructor help page under the `session` section and I have updated the file  `src/main/webapp/partials/instructorHelpSessions.jsp` accordingly.  I have added the screenshots of the UI changes in the instructorHelp page in this comment : https://github.com/TEAMMATES/teammates/pull/8265#issue-284903396
